### ISSUE TITLE
Unable to add Allied Telesis AT-28GTX with IXP Manager v4. 

### ISSUE THREAD

**Issue by zeeshanaziz001** [inex/IXP-Manager#393]
Hello!
         I am trying to integrate **Allied Telesis AT-28GTX** switch with **IXP Manager v4** through **SNMP v2c** but unable to become successful in it. Actually there is no support for Allied Telesis currently in IXP Manager. I have followed your guide on how to add platform support by going through these links.

https://github.com/opensolutions/OSS_SNMP/wiki/Device-Discovery
https://github.com/opensolutions/OSS_SNMP/blob/master/examples/platform.php

I have copied the example script from the above link and paste it on this path  **/srv/ixpmanager/vendor/opensolutions/oss-snmp/src/OSS_SNMP/Platforms/**
and named it as vendor_allied_telesis.php. The machine on which i am running IXP Manager v4 is **Ubuntu Server 16.04.3 LTS** codename **xenial**. I have configured SNMP agent on the switch using the official documentation of Allied Telesis. When I try to add the the switch from the IXP Manager GUI I get this error:

Uh oh! Unexpected Error!
We're very sorry but something we didn't expect has just happened.

Following are the outputs of the commands that i have run on Machine where ixp manager is installed:
sudo snmpget -c xbits -On -v 2c 192.168.100.170 .1.3.6.1.2.1.1.1.0
**Output:**
.1.3.6.1.2.1.1.1.0 = STRING: "Allied Telesis router/switch, AW+ v5.4.7-2.5"

sudo snmpget -c xbits -On -v 2c 192.168.100.170 .1.3.6.1.2.1.1.2.0
Output:
.1.3.6.1.2.1.1.2.0 = OID: .1.3.6.1.4.1.207.1.14.118

Management Interface IP = 192.168.100.170
Community String = xbits

I am unable to find the the log location of IXP Manager as well so suggest me how to add support for the Allied Telesis switch. The screenshot of error is attached below for reference.

![image](https://user-images.githubusercontent.com/37068949/36971511-db9973de-208d-11e8-85e5-8786e77c134d.png)

**Comment by barryo** [comment]
You should see the error if you set `APP_DEBUG=true` in the `.env` file.

Otherwise, you can tail (adjusting path for your system) the following:

```sh
tail -f /srv/ixpmanager/storage/logs/laravel.log
```

**Comment by zeeshanaziz001** [comment]
Hi Barryo!
                Appreciate your response. Let me enable the debug flag and i will share the logs with you.
Thanks.

**Comment by zeeshanaziz001** [comment]
Hi!
    Below is the log that is generated when i try to add the switch by snmp.

2018-03-05T11:09:34+00:00 DEBUG (7) [192.168.100.12]: Logger instantiated
2018-03-05T11:09:34+00:00 DEBUG (7) [192.168.100.12]:

2018-03-05T11:09:34+00:00 DEBUG (7) [192.168.100.12]: ErrorController::errorAction()
2018-03-05T11:09:34+00:00 ERR (3) [192.168.100.12]:

************************************************************************
****************************** EXCEPTIONS *******************************
************************************************************************

--------------------------- EXCEPTION --------------------------

Message: Undefined variable: argv
Line: 33
File: /srv/ixpmanager/vendor/opensolutions/oss-snmp/src/OSS_SNMP/Platforms/vendor_allied_telesis.php

**Comment by barryo** [comment]
Okay, so it looks like you have an issue in your code:

```
Message: Undefined variable: argv
Line: 33
File: /srv/ixpmanager/vendor/opensolutions/oss-snmp/src/OSS_SNMP/Platforms/vendor_allied_telesis.php
```

**Comment by zeeshanaziz001** [comment]
Yes. I have put your example script in this file from the below link.
https://github.com/opensolutions/OSS_SNMP/blob/master/examples/platform.php
Do I need to modify it?

**Comment by barryo** [comment]
>  Do I need to modify it?

That's a command line script for testing.

Can you follow [these details](https://github.com/opensolutions/OSS_SNMP/wiki/Device-Discovery#adding-new-devices) and open an issue over on the https://github.com/opensolutions/OSS_SNMP project?

**Comment by zeeshanaziz001** [comment]
Ok I will open an issue over there.
Thanks

**Comment by barryo** [comment]
This was implemented by OSS_SNMP via opensolutions/OSS_SNMP@4c67c2778d81465e73badbbf56864d874007d0dd but it won't make its way into IXP Manager until the next release which is not scheduled until later this month.

To use it in the mean time, proceed as follows:

```sh
# cd into the root of your IXP Manager installation (${IXPROOT}) such as:
$ cd /srv/ixpmanager

# then move to the OSS_SNMP vendor directory
$ cd vendor/opensolutions/oss-snmp

# then update to the latest release (and you should see something like what follows):
$ git pull
remote: Counting objects: 22, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 22 (delta 9), reused 17 (delta 9), pack-reused 5
Unpacking objects: 100% (22/22), done.
From https://github.com/opensolutions/OSS_SNMP
   a2af3cb..4c67c27  master     -> origin/master
Updating 6b7afc3..4c67c27
Fast-forward
 src/OSS_SNMP/Platforms/vendor_allied-telesis.php     | 59 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 src/OSS_SNMP/Platforms/vendor_cisco.php              | 16 +++++++++-------
 tests/Tests/OSS_SNMP/Platforms/AlliedTelesisTest.php | 26 ++++++++++++++++++++++++++
 tests/Tests/OSS_SNMP/Platforms/CiscoTest.php         | 13 ++++++++++++-
 4 files changed, 106 insertions(+), 8 deletions(-)
 create mode 100644 src/OSS_SNMP/Platforms/vendor_allied-telesis.php
 create mode 100644 tests/Tests/OSS_SNMP/Platforms/AlliedTelesisTest.php
```

You will also need to create a new vendor in IXP Manager. While it must be called `Allied Telesis`, we would suggest you create it with the following values:

![screen shot 2018-03-06 at 17 39 49](https://user-images.githubusercontent.com/804649/37048316-6a12fff4-2165-11e8-9a1a-0f7620069704.png)

**Comment by zeeshanaziz001** [comment]
Hi!     Hope you are fine. Really appreciate your response and suggestion. I will try this method and let you know if it works successfully.
Regards:Zeeshan Aziz

Sent from the Yahoo Mail app. Get yours! 
 
  On Tue, 6 Mar 2018 at 10:41 pm, barryo<notifications@github.com> wrote:   
This was implemented by OSS_SNMP via opensolutions/OSS_SNMP@4c67c27 but it won't make its way into IXP Manager until the next release which is not scheduled until later this month.

To use it in the mean time, proceed as follows:
# cd into the root of your IXP Manager installation (${IXPROOT}) such as:
$ cd /srv/ixpmanager

# then move to the OSS_SNMP vendor directory
$ cd vendor/opensolutions/oss-snmp

# then update to the latest release (and you should see something like what follows):
$ git pull
remote: Counting objects: 22, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 22 (delta 9), reused 17 (delta 9), pack-reused 5
Unpacking objects: 100% (22/22), done.
From https://github.com/opensolutions/OSS_SNMP
   a2af3cb..4c67c27  master     -> origin/master
Updating 6b7afc3..4c67c27
Fast-forward
 src/OSS_SNMP/Platforms/vendor_allied-telesis.php     | 59 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 src/OSS_SNMP/Platforms/vendor_cisco.php              | 16 +++++++++-------
 tests/Tests/OSS_SNMP/Platforms/AlliedTelesisTest.php | 26 ++++++++++++++++++++++++++
 tests/Tests/OSS_SNMP/Platforms/CiscoTest.php         | 13 ++++++++++++-
 4 files changed, 106 insertions(+), 8 deletions(-)
 create mode 100644 src/OSS_SNMP/Platforms/vendor_allied-telesis.php
 create mode 100644 tests/Tests/OSS_SNMP/Platforms/AlliedTelesisTest.php
You will also need to create a new vendor in IXP Manager. While it must be called Allied Telesis, we would suggest you create it with the following values:
